File: clerk-front/src/model/Project.ts

export interface Project {
    id: number;
    name: string;
    createTime: Date;
    comment: string | null;
}

export interface RawProject {
    id: number;
    name: string;
    createTime: string;
    comment: string | null;
}

export interface NewProject {
    name: string;
    comment: string | null;
}

export function fromRaw(raw: RawProject): Project {
    const date = new Date(raw.createTime + 'Z');
    return {
        ...raw,
        createTime: date,
    }
}

File: Clerk/Model/Project.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;


namespace Clerk.Model {

    public class Project {
        
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        
        [Required]
        public string Name { get; set; }
        
        [Required]
        public string OwnerId { get; set; }
        
        [JsonIgnore]
        [Required]
        public ClerkUser Owner { get; set; }
        
        public string Comment { get; set; }
        public DateTime CreateTime { get; set; }
        
        [JsonIgnore]
        public List<Experiment> Experiments { get; set; }
        
    }

    public class NewProject
    {
        [Required]
        public string Name { get; set; }
        
        public string Comment { get; set; }
    }
}